### PR TITLE
Attendance Tracking: Attendees

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Context: [Spec](https://docs.google.com/document/d/1hU2SQvnfkfxu3RoxhRCKL6IuWl6zO7nuZFEbgF5grrE/edit?pli=1#heading=h.whks7x3gtv74)

Adds UI to list expected attendees for an Attendance Tracking Event.

image

I kept "Add `remove_event()` view" and "Drop `remove_event()` view" commits so that we can revert the "Drop" commit if we want that option. It might be useful to be able to delete an Event under limited circumstances?

:blowfish: 

## Feature Flag

Attendance Tracking

### Safety story

Tested locally

### Automated test coverage

Includes tests

### QA Plan

QA planned when feature is more mature.

See comment [here](https://github.com/dimagi/commcare-hq/pull/32705#issuecomment-1463661423).

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Charl1996** [comment]
@kaapstorm Are we continuing to use the `username` column?

**Comment by kaapstorm** [comment]
Good question @Charl1996 . I left the columns as they were, but maybe that should change. Do you think we should include it for attendees who were created from users, but maybe make it the third column? Or would that be confusing when attendees are beneficiaries / not users?

**Comment by Charl1996** [comment]
> Good question @Charl1996 . I left the columns as they were, but maybe that should change. Do you think we should include it for attendees who were created from users, but maybe make it the third column? Or would that be confusing when attendees are beneficiaries / not users?

Maybe we can leave it out then? I'm just wondering again what the unique identifier will be for the attendees.

File: corehq/apps/events/models.py
Comment (Charl1996) [reply corehq/apps/events/models.py:30]: Good point. I think we should reserve this.

File: corehq/apps/events/models.py
Comment (Charl1996) [reply corehq/apps/events/models.py:113]: I don't think it will ever be `CommCareCase` instances when we pass in the data directly from the frontend, but I think it also has to do with the expectation. When I read a method like `set_expected_attendees` I would expect to be able to provide a list of `CommCareCase` instances, since an attendee is a `CommCareCase` instance. There might exist a use-case in the future where we have a list of attendee cases, in which case we either strip out the id's beforehand or simply have this method do it automatically for us. Having this is a nice-to-have at the moment I think, but it keeps future code clean.

File: corehq/apps/events/views.py
Comment (zandre-eng) [inline corehq/apps/events/views.py:261]: Will this be done in this PR?

File: corehq/apps/events/models.py
Comment (kaapstorm) [reply corehq/apps/events/models.py:30]: The prefix "commcare-" was chosen so that app builders are unlikely to _accidentally_ name a case type with the same name. But they should be able to do it _intentionally_.

The reason we didn't make this reserved (the way we do for "commcare-user") is a _allow_ Event Managers to import attendees. (You can't import user cases in HQ.) The spec says: 

> Web users can use Case Import to import attendee cases. They must have “commcare-attendee” as their case type.

We also want app builders to be able to build and customize apps that create and manage attendee cases. The spec goes on,

> The Attendees page uses CRUDPaginatedView to list cases of case type “commcare-attendee” by name. It offers the ability to add an attendee case. The form only prompts for the “name” case property.

The reason we only assume the existence of the "name" property is because we want to be flexible enough to allow projects to decide the schema of their own attendees. Are they mobile workers? Are they farmers? Students? Kids at a day care? We should not be nailing that down. Each project should be able to build attendance tracking around their own use case.

**Comment by zandre-eng** [comment]
PR looks good. Given @kaapstorm's response to a previous comment, I am happy to approve once the todo is handled :+1:

**Comment by Charl1996** [comment]
@kaapstorm 
I think this PR wants a QA pass, but at the same time QA would be a little blocked on not being able to create attendees (hence cannot really fully test the events logic).

Do you think we should hold off on merging this PR to master till your work is on staging for QA, or go ahead and merge and only QA once your work is in?

**Comment by kaapstorm** [comment]
Offlined with @Charl1996 . We reckon that because all this is behind a feature flag, and because none of the new code is used outside `corehq.apps.events`, it's OK to merge. The longer more that one of us are working in this area, the more merge conflicts we're going to have to resolve, and it would be nice to minimize that. Once we have enough functionality for QA to test properly (specifically adding attendees) we can hand it over to them.

I think we're ready for a final review.

**Comment by zandre-eng** [comment]
LGTM once the unused import is removed. :+1:

**Review by sravfeyn — APPROVED** [review]

**Pull request merged**
